### ISSUE TITLE
FireFox Support

### ISSUE THREAD

**Issue by Unthrottled** [doki-theme/doki-theme-chrome#14]
We need fire fox support!

**Comment by Unthrottled** [comment]
Started because of #13

**Comment by Unthrottled** [comment]
@ZimCodes feel free to use this a channel of communication to things related to the fire fox integration. Thanks again!

**Comment by ZimCodes** [comment]
Understood. For the future, are there any plans for constructing a browser compatible extension?

To clarify a little further, are there plans for making one browser extension that's compatible with all browsers? Or keeping how it is now. One extension per browser. Each providing their own customary feature set?

**Comment by Unthrottled** [comment]
> For the future, are there any plans for constructing a browser compatible extension?

I don't know if that is possible, each browser behaves a bit differently (at least from my experience.) If it is possible to only maintain a master web extension, then that would be awesome.

However, if it is not, then I'll probably just leave a separate `chrome`, `firefox`, and `edge` setup. Right now it's more convenient to keep them all in the same place, so the buildscript doesn't have to be moved around and changed too much.

**Comment by Unthrottled** [comment]
@ZimCodes 
I've started on getting the firefox extension migrated over to the build process today (I wasn't able to start yesterday like I hoped :disappointed:  ).

I've made some headway, I've got most of it done, I'll see how much I can get done later today. I've got some weird edge cases with color conversion on the light and dark themes.

I left most of the extension code intact. I just needed to strategically plugin the theme definitions. While I appreciate what you did with overall look and feel.
I did change the look and feel to be more consistent with the products on the other platforms. 

![Screenshot from 2020-12-21 12-45-41](https://user-images.githubusercontent.com/15972415/102811136-8a505380-438a-11eb-809b-a101e1090e19.png)

You can see all my changes here: https://github.com/doki-theme/doki-theme-chrome/compare/5fb95b55e3a9d757a92332ec352a64cc00b0ed50...firefox

**Comment by Unthrottled** [comment]
@ZimCodes Good news! I was able to get it completed! [You can see the release](https://github.com/doki-theme/doki-theme-chrome/releases/tag/v6.0.0) here.

Thank you for your work and enabling me to get this completed in a day!

![Thank you](https://media.giphy.com/media/fWfowxJtHySJ0SGCgN/giphy.gif)

Firefox still has some quirks, just like the other browsers, but I think Firefox has more potential.

I'll have to look into publishing the extension sometime later, so I don't have to keep adding it to the browser. However, for now, I'll be focusing on [adding some more waifu](https://github.com/doki-theme/doki-master-theme/milestone/1) (as if 31 wasn't enough :sweat_smile: )

**Comment by Unthrottled** [comment]
Oh yeh, feel free to keep submitting contributions to master if you want :smile:

**Comment by ZimCodes** [comment]
Nice! That was quick. I tested out the Firefox extension in the 6.0.0 release & i got to say... I dig it. For a moment, I thought the popup menu changing colors like that was a bug, but it was a feature. There are some bugs in there. Nothing a little typing can't fix.

In any case, I finished reworking the method for loading the background images onto a new tab. This rework will provide smoother/faster transitions, as this feature takes advantage of the _chrome_url_override_ key to load a custom page rather than forcing the browser to redirect there.
